alacritty/src/macos/proc.rs

```diff
@@ -0,0 +1,117 @@
+use std::os::raw;
+use std::path::PathBuf;
+
+pub fn cwd(pid: raw::c_int) -> io::Result<PathBuf> {
+    let mut info = MaybeUninit::<sys::proc_vnodepathinfo>::uninit();
+    let info_ptr = info.as_mut_ptr() as *mut raw::c_void;
+    let size = mem::size_of::<sys::proc_vnodepathinfo>() as raw::c_int;
+
+    let c_str = unsafe {
+        let pidinfo_size = sys::proc_pidinfo(pid, sys::PROC_PIDVNODEPATHINFO, 0, info_ptr, size);
+        match pidinfo_size {
+            c if c < 0 => return Err(io::Error::last_os_error()),
+            s if s != size => {
+                return Err(io::Error::new(io::ErrorKind::Other, "invalid return size"))
+            },
+            _ => CStr::from_ptr(info.assume_init().pvi_cdir.vip_path.as_ptr()),
+        }
+    };
+
+    CString::from(c_str)
+        .into_string()
+        .map(PathBuf::from)
+        .map_err(|e| io::Error::new(io::ErrorKind::Other, e))
+}
+
+// Cleaned up `bindgen` output below
+#[allow(non_camel_case_types)]
+mod sys {
+    use std::os::raw;
+
+    pub const PROC_PIDVNODEPATHINFO: raw::c_int = 9;
+
+    type gid_t = raw::c_int;
+    type off_t = raw::c_longlong;
+    type uid_t = raw::c_int;
+    type fsid_t = fsid;
+
+    #[repr(C)]
+    #[derive(Debug, Copy, Clone)]
+    pub struct fsid {
+        pub val: [i32; 2usize],
+    }
+
+    #[repr(C)]
+    #[derive(Debug, Copy, Clone)]
+    pub struct vinfo_stat {
+        pub vst_dev: u32,
+        pub vst_mode: u16,
+        pub vst_nlink: u16,
+        pub vst_ino: u64,
+        pub vst_uid: uid_t,
+        pub vst_gid: gid_t,
+        pub vst_atime: i64,
+        pub vst_atimensec: i64,
+        pub vst_mtime: i64,
+        pub vst_mtimensec: i64,
+        pub vst_ctime: i64,
+        pub vst_ctimensec: i64,
+        pub vst_birthtime: i64,
+        pub vst_birthtimensec: i64,
+        pub vst_size: off_t,
+        pub vst_blocks: i64,
+        pub vst_blksize: i32,
+        pub vst_flags: u32,
+        pub vst_gen: u32,
+        pub vst_rdev: u32,
+        pub vst_qspare: [i64; 2usize],
+    }
+
+    #[repr(C)]
+    #[derive(Debug, Copy, Clone)]
+    pub struct vnode_info {
+        pub vi_stat: vinfo_stat,
+        pub vi_type: raw::c_int,
+        pub vi_pad: raw::c_int,
+        pub vi_fsid: fsid_t,
+    }
+
+    #[repr(C)]
+    #[derive(Copy, Clone)]
+    pub struct vnode_info_path {
+        pub vip_vi: vnode_info,
+        pub vip_path: [raw::c_char; 1024usize],
+    }
+
+    #[repr(C)]
+    #[derive(Copy, Clone)]
+    pub struct proc_vnodepathinfo {
+        pub pvi_cdir: vnode_info_path,
+        pub pvi_rdir: vnode_info_path,
+    }
+
+    extern "C" {
+        pub fn proc_pidinfo(
+            pid: raw::c_int,
+            flavor: raw::c_int,
+            arg: u64,
+            buffer: *mut raw::c_void,
+            buffersize: raw::c_int,
+        ) -> raw::c_int;
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::env;
+    use std::process;
+
+    #[test]
+    fn test_cwd() {
```

Comment:
We don't need to call the test a test in the method name. Either we call this just `fn cwd` or we actually describe what about `cwd` it tests. Like `cwd_matches_std` or something like that.

Reply:
Well, the point is to avoid name collision afaics, but I can rename test.

Reply:
It's a separate module, there is no collision.

Reply:
No, the function from top module is called `cwd`, then they do `super::*`, which brings cwd, here, and then you call a test `cwd`, there's a collision due to the way it's used. So to avoid that you should either rename test or use full path for function, like `super::cwd`.

Reply:
yea IIRC I started with this in the same module. Renaming the test just `cwd` would even be fine with me.

Reply:
Just to be clear, even when `super::*` is used, there is no problem with conflicts. But of course @nixpulvis suggestion is not unlikely, if it wasn't in a separate module it would be problematic.